diff --git a/lib/resolveModuleName.js b/lib/resolveModuleName.js
index 209030033139abe7ff2fafc7d071ed385f5047b4..e36684452de21c16034ac5a72630aebd15e4a663 100644
--- a/lib/resolveModuleName.js
+++ b/lib/resolveModuleName.js
@@ -84,11 +84,7 @@ function createResolveModuleName(ts, getFileSize, host, languagePlugins, getSour
     };
     // fix https://github.com/vuejs/language-tools/issues/3332
     function fileExists(fileName) {
-        if (host.fileExists(fileName)) {
-            const fileSize = getFileSize?.(fileName) ?? host.readFile(fileName)?.length ?? 0;
-            return fileSize < 4 * 1024 * 1024;
-        }
-        return false;
+        return host.fileExists(fileName);
     }
 }
 //# sourceMappingURL=resolveModuleName.js.map
